File: QTD.SPA/src/app/_DtoModels/Instance/InstanceSetting.ts
import { IdentityProviderVM } from '../IdentityProvider/IdentityProviderVM';
import { Entity } from '../Entity';
import { Instance } from './Instance';

export class InstanceSetting extends Entity {
  id: number;
  instanceId: number;
  databaseName: string;
  databaseVersion: string;
  instance: Instance;
  scormTenant: string;
  clientAccountNumber: string;
  defaultIdentityProvider:IdentityProviderVM
  defaultIdentityProviderId:string;
  mfaEnabled:boolean;
  publicUrl: string;
}

File: QTD2.Domain/Entities/Authentication/InstanceSetting.cs
using QTD2.Domain.Entities.Common;

namespace QTD2.Domain.Entities.Authentication
{
    public class InstanceSetting : Entity
    {
        public int InstanceId { get; set; }

        public string DatabaseName { get; set; }

        public string DataBaseVersion { get; set; }
        public bool? MFAEnabled { get; set; }
        public virtual Instance Instance { get; set; }

        public string ScormTenant { get; set; }
        public int ClientAccountNumber { get; set; }
        public int? DefaultIdentityProviderId { get; set; }
        public string PublicUrl {  get; set; }
        public virtual IdentityProvider DefaultIdentityProvider { get; set; }

        public InstanceSetting(int instanceId, string databasename, string databaseVersion)
        {
            InstanceId = instanceId;
            DatabaseName = databasename;
            DataBaseVersion = databaseVersion;
        }

        public InstanceSetting()
        {
        }

        private void UpdateVersion(string version)
        {
        }
    }
}
